sdk/eventhubs/azure-messaging-eventhubs/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,269 @@
+  - [Setting partition key on EventData is not set in Kafka consumer](#setting-partition-key-on-eventdata-is-not-set-in-kafka-consumer)
+- [Troubleshoot EventProcessorClient issues](#troubleshoot-eventprocessorclient-issues)
+  - [412 precondition failures when checkpointing](#412-precondition-failures-when-checkpointing)
+  - [Partition ownership changes a lot](#partition-ownership-changes-a-lot)
+  - ["...current receiver 'nil' with epoch '0' is getting disconnected"](#current-receiver-nil-with-epoch-0-is-getting-disconnected)
+  - [High CPU usage](#high-cpu-usage)
+  - [Processor client stops receiving](#processor-client-stops-receiving)
+  - [Migrate from legacy to new client library](#migrate-from-legacy-to-new-client-library)
+- [Get additional help](#get-additional-help)
+
+## Handle Event Hubs exceptions
+
+All Event Hubs exceptions are wrapped in an [AmqpException][AmqpException].  They often have an underlying AMQP error code which specifies whether an error is retryable or non-retryable.  For retryable errors (ie. "amqp:connection:forced" or "amqp:link:detach-forced"), the client libraries will recover from these errors based on the [_retry options_][AmqpRetryOptions] set when instantiating the client.  To configure retry options, follow the sample [Publish events to specific partition][PublishEventsToSpecificPartition].  If the error is non-retryable, there is some configuration issue that the customer needs to resolve.
+
+The recommended way to solve the specific exception the AMQP exception represents is to follow the
+[Event Hubs Messaging Exceptions][event_hubs_messaging_exceptions] guidance.
+
+### Find relevant information in exception messages
+
+An [AmqpException][AmqpException] contains 3 fields which describe the error.
+
+* **getErrorCondition**: The underlying AMQP error. A description of the errors can be found in the [AmqpErrorCondition][AmqpErrorCondition] javadocs or the OASIS AMQP 1.0 spec.
+* **isTransient**: Whether or not trying to perform the same operation is possible.  SDK clients apply the retry policy when the error is transient.
+* **getErrorContext**: Information about where the AMQP error originated.
+  * [LinkErrorContext][LinkErrorContext]: Errors that occur in either the send/receive link.
+  * [SessionErrorContext][SessionErrorContext]: Errors that occur in the session.
+  * [AmqpErrorContext][AmqpErrorContext]: Errors that occur in the connection or a general AMQP error.
+
+### Commonly encountered exceptions
+
+#### amqp:connection:forced and amqp:link:detach-forced
+
+When the connection to Service Bus or Event Hubs is idle, after a period of time, the service will disconnect the client.  This is not a problem as the clients will reestablish a connection with the service. More information: https://docs.microsoft.com/azure/service-bus-messaging/service-bus-amqp-troubleshoot
```

Comment:
nit: Does Service Bus need to be mentioned here?

second nit: Do we want to use the full URL here or use a decorative link?

Reply:
Along those same lines - I recall @jsquire  has mentioned _different_ limits when it comes to Event Hubs so it might be that the SB limits are not the same?